internal/server/cache/cache.go

```diff
@@ -0,0 +1,67 @@
+package cache
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-redis/redis/v8"
+	rate "github.com/go-redis/redis_rate/v9"
+
+	"github.com/infrahq/infra/internal/logging"
+)
+
+type Cache struct {
+	client *redis.Client
+}
+
+type Options struct {
+	Host string
+	Port int
+	Username string
+	Password string
+	Options string
+}
+
+func NewCache(options Options) *Cache {
+	var client *redis.Client
+
+	if len(options.Host) > 0 {
+		redisOptions, err := redis.ParseURL(fmt.Sprintf("redis://%s:%d?%s", options.Host, options.Port, options.Options))
+		if err != nil {
+			logging.Warnf("invalid cache options: %#v", options)
+			return nil
+		}
```

Comment:
Only logging a warning seems like the wrong behaviour here. If the operator specifies any of these options then they are expecting redis to be configured. I think we should return the error.

```suggestion
		if err != nil {
			return nil, fmt.Errorf("invalid redis configuration: %w", err)
		}
```

Reply:
You're probably right. A misconfiguration should fail fast.